subcommand/inject-connect/command_test.go

```diff
@@ -188,53 +202,21 @@ func TestRun_ResourceLimitDefaults(t *testing.T) {
 }
 
 func TestRun_ValidationConsulHTTPAddr(t *testing.T) {
-	cases := []struct {
-		name    string
-		envVars []string
-		flags   []string
-		expErr  string
-	}{
-		{
-			envVars: []string{api.HTTPAddrEnvName, "%"},
-			flags: []string{"-consul-k8s-image", "hashicorp/consul-k8s", "-consul-image", "foo", "-envoy-image", "envoy:1.16.0",
-				"-enable-health-checks-controller=true"},
-			expErr: "Error parsing consul address \"http://%\": parse \"http://%\": invalid URL escape \"%",
-		},
-	}
-	for _, c := range cases {
-		t.Run(c.expErr, func(t *testing.T) {
-			k8sClient := fake.NewSimpleClientset()
-			ui := cli.NewMockUi()
-			cmd := Command{
-				UI:        ui,
-				clientset: k8sClient,
-			}
-			os.Setenv(c.envVars[0], c.envVars[1])
-			code := cmd.Run(c.flags)
-			os.Unsetenv(c.envVars[0])
-			require.Equal(t, 1, code)
-			require.Contains(t, ui.ErrorWriter.String(), c.expErr)
-		})
-	}
-}
-
-// Test that with health checks enabled, if the listener fails to bind that
-// everything shuts down gracefully and the command exits.
-func TestRun_CommandFailsWithInvalidListener(t *testing.T) {
 	k8sClient := fake.NewSimpleClientset()
 	ui := cli.NewMockUi()
 	cmd := Command{
 		UI:        ui,
 		clientset: k8sClient,
 	}
-	code := cmd.Run([]string{
-		"-consul-k8s-image", "hashicorp/consul-k8s", "-consul-image", "foo", "-envoy-image", "envoy:1.16.0",
-		"-enable-health-checks-controller=true",
-		"-http-addr=http://0.0.0.0:9999",
-		"-listen", "999999",
-	})
+	flags := []string{"-consul-k8s-image", "hashicorp/consul-k8s", "-consul-image", "foo", "-envoy-image", "envoy:1.16.0",
+		"-enable-health-checks-controller=true"}
+
+	os.Setenv(api.HTTPAddrEnvName, "%")
+	code := cmd.Run(flags)
```

Comment:
Did this work in CI? I think this would run into the same issue I was running into, where there is no kubeconfig for the manager.

I learned from @lkysow  that the original purpose of the `TestRun_CommandFailsWithInvalidListener` test was to make sure that when there was an error that the whole command would actually exit rather than keep on running (apparently there was a bug/issue where the healthchecks controller would keep running).

Making sure that the command exits on validation errors still seems to satisfy that purpose and makes sense to me. So, if we can do that while skipping the manager itself running (if the manager causes this test to fail) then  that would be good. Currently, the only way I can think of doing that is adding a conditional around running the manager, and simply not running it during tests that do `cmd.Run`

Reply:
Oh right I realize you can't test this in CI until after you can rebase on endpoints controller. Can we take a closer look at this test after the rebase and make sure it's testing the original purpose well, and that it passes?

Reply:
That sounds good! This diff definitely makes it harder to look at. The validation test for the invalid listener now fails before starting the manager because the webhook runs in the same process as the manager does but the validation error occurs before the manager starts and hence does run successfully in CI. 

`inject-connect/command.go` L386-395 capture this validation error and they will return before the manager starts.

Reply:
Ah I see, that makes sense!